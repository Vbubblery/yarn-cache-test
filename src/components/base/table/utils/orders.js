import { frozenPeriods, isFrozen, isMoq, isLotSize } from "./planning";
import { defaultTableValueFormatting, isNotPast } from "./base.js";

export const ORDER_STATUS = {
  VALIDATED: "validated",
  LOCKED: "locked"
};
export const ORDER_FREEZE_STATUS = {
  PARTIAL: "partlyFrozen",
  TOTAL: "fullyFrozen"
};

export const getOrderStatus = status => {
  switch (status) {
    case 2:
      return ORDER_STATUS.VALIDATED;
    case 3:
    case 4:
      return ORDER_STATUS.LOCKED;
    default:
      return undefined;
  }
};

export const getOptionsForOrders = options => {
  const { leadTimes, periods, periodType, productData } = options;
  return {
    periodType,
    periods,
    frozenPeriods: frozenPeriods(
      productData?.leadTime || {},
      periods,
      periodType,
      leadTimes
    ),
    isAggregated: options.isAggregated,
    access: options.access,
    leadTimes: options.leadTimes,
    units: options.productData?.units,
    moq: options.productData?.moq,
    lotSize: options.productData?.lotSize
  };
};

export const getOptionsForStatus3Supplies = options => {
  const { periods, periodType, isAggregated, access } = options;
  return {
    periodType,
    periods,
    isAggregated,
    access,
    units: options.productData?.units
  };
};

export const getOptionsForDemand = options => ({
  periodType: options.periodType,
  periods: options.periods,
  isAggregated: options.isAggregated,
  access: options.access
});

export const getOrdersValues = (orders, options) => {
  if (orders.isParent) {
    return orders.values.map((value, index) => ({
      value,
      isPast: index === 0
    }));
  }

  const {
    isAggregated,
    moq: moqBySupplier,
    lotSize: lotSizeBySupplier,
    ...otherOptions
  } = options;
  const { editedBy, orderStatus, supplier, isParent } = orders;
  return orders.values.map((value, index) => {
    const status = orderStatus && getOrderStatus(orderStatus[index]);
    const isByFlowlity = editedBy && editedBy[index] === "flowlity";
    const frozenStatus = isFrozen(index, supplier, options);
    const moq = isMoq(supplier, isAggregated, orders, moqBySupplier);
    const lotSize = isLotSize(
      supplier,
      isAggregated,
      orders,
      lotSizeBySupplier
    );
    return {
      value,
      isByFlowlity,
      status,
      moq,
      lotSize,
      isPast: index === 0,
      tooltip: supplier !== null || supplier !== undefined,
      isEditable: options.access && !isAggregated && !isParent && index > 0,
      ...(!isNaN(value) && { isFrozen: frozenStatus }),
      ...otherOptions
    };
  });
};

export const getStatus3SuppliesValues = (supplies, options) => {
  const { isAggregated, access, ...otherOptions } = options;
  return supplies.values.map((value, index) => {
    return {
      value,
      isPast: index === 0,
      isEditable: access && !isAggregated && index > 0,
      ...otherOptions
    };
  });
};

export const getAllOrdersValues = (allOrders, options) => {
  return allOrders.values.map((value, index) => {
    return {
      value,
      isPast: !isNotPast(index, options.periods, options.periodType)
    };
  });
};

export const allOrdersCalc = (value, isPast) => {
  if (isPast) return null;
  else if (!value) return 0;
  return defaultTableValueFormatting(value);
};
