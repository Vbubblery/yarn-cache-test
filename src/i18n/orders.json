{
  "en": {
    "orderModalAddProductsButton": "Add products",
    "constraintsNotMetMessage": "Some of the products don't meet their constraints",
    "moq": "Minimum order quantity",
    "constraint": "Constraint | Constraint | Constraints",
    "orderConstraints": "Order constraints",
    "orderModalTitle": "Order for",
    "editingModalTitle": "Editing order for",
    "orderTitle": "Orders",
    "planned": "Purchase requisition | Purchase requisition | Purchase requisitions",
    "validated": "Confirmed purchase requisition | Confirmed purchase requisition | Confirmed purchase requisitions",
    "purchaseOrders": "Purchase order | Purchase order | Purchase orders",
    "salesOrders": "Sales order | Sales order | Sales orders",
    "delivered": "Delivered | Delivered | Delivered",
    "search": "Search by order ID, supplier or product",
    "ship_before": "Ship before",
    "send_before": "Send before",
    "date": "Delivery date",
    "totalQuantity": "Total quantity",
    "quantity": "Quantity",
    "addedItems": "Added items",
    "groupedBy": "Grouped by",
    "inTotal": "In total",
    "download_selected": "Download",
    "upload_selected": "Send to API",
    "sftp_selected": "Export to SFTP",
    "sftp_exported": "Exported to SFTP",
    "temporary_id": "Temporary ID",
    "order_id": "Order ID",
    "select_all": "Select all",
    "deselect_all": "Deselect all",
    "deleteSelected": "Delete selected",
    "noOrders": "No orders found",
    "delivery": "Delivery",
    "constraintFrequency": "Doesn't meet order frequency requirements",
    "constraintMOQ": "Doesn't meet MOQ requirements",
    "constraintFullTruck": "Doesn't meet full truck requirements",
    "issueMOQ": "product have issue with MOQ or lot size | products have issue with MOQ or lot size",
    "orderIssues": "Order issues",
    "productIssues": "Product issues",
    "days": "days",
    "latestOrderDate": "Latest Order Date",
    "deliveryDate": "Delivery Date",
    "deselectAll": "Deselect all",
    "suppliers": "Suppliers",
    "reset": "Reset",
    "dateRange": "Date range",
    "thisWeek": "This week",
    "in2Weeks": "In 2 weeks",
    "in1month": "In 1 month",
    "in2months": "In 2 months",
    "in3months": "In 3 months",
    "tempId": "Temporary ID",
    "orderTitleDetail": "Details",
    "orderTitleStatus1": "Purchase requisition",
    "orderTitleStatus2": "Confirmed purchase requisition",
    "orderTitleStatus3": "Purchase order",
    "orderTitleStatus4": "Delivered order",
    "orderDetailMessageTitle": "Messages",
    "orderId": "Order ID",
    "day": "day | day | days",
    "value": "Value",
    "product": "Product",
    "reference": "Reference",
    "lotSize": "Lot Size",
    "leadTime": "Delivery lead time",
    "daysOfCoverage": "Stock coverage at delivery",
    "goodMoq": "Min. order quantity",
    "orderFrequency": "Order frequency",
    "badFullTruckQuantity": "The order is not respecting the full truck quantity constraint",
    "badOrderFrequency": "The order is not respecting the frequency constraint",
    "fullTruckQty": "Full truck quantity",
    "badMOQSupply": "The supply is not respecting the MOQ constraint",
    "badMOQOrder": "The order is not respecting the MOQ constraint",
    "expectedDelivery": "Expected delivery",
    "delete": "Delete",
    "cancel": "Cancel",
    "edit": "edit",
    "cancelEdit": "cancel edit",
    "discardChanges": "Discard Changes",
    "save": "Save",
    "orderIsUpdating": "Order is updating...",
    "deleteSuccess": "Order has been deleted",
    "deleteOrderError": "Cannot delete the order",
    "deleteSupplyError": "The product could not be deleted correctly",
    "deleteWarningTitle": "Delete confirmation",
    "deleteCurrentOrderWarningText": "Are you sure you want to remove this order?",
    "deleteCurrentOrderSupplyWarningText": "This will delete all the planned supplies inside this order.",
    "deleteSupplyWarningText": "Are you sure you want to remove this product ?",
    "deleteWarningConfirmButton": "YES, CONFIRM",
    "deleteWarningCancelButton": "CANCEL",
    "mergeWarningTitle": "Merge confirmation",
    "mergeWarningText": "An order already exists for this date and supplier. Do you want to combine the two orders?\n This will add the current products and quantities to the order found at this new date.",
    "mergeWarningConfirmButton": "YES, CONFIRM",
    "mergeWarningCancelButton": "CANCEL",
    "secondaryUnit": "Secondary unit",
    "orderAndSupplyConstraintNotMet": "Some constraints on the order level and product level are not met",
    "orderConstraintNotMet": "Some constraints on the order level are not met",
    "supplyConstraintNotMet": "Some products don't meet their constraints",
    "customer": {
      "display": "Customer",
      "partner": "Supplier",
      "waitValidateOrder": "Waiting to be validated",
      "orderValidated": "Validated",
      "waitSendOrder": "Waiting to be sent",
      "orderSent": "Sent",
      "sendBefore": "Send before",
      "badLotSize": "Lot size not respected",
      "validateBtn": "Validate"
    },
    "supplier": {
      "display": "Supplier",
      "partner": "Customer",
      "waitValidateOrder": "Waiting to be accepted",
      "orderValidated": "Accepted",
      "waitSendOrder": "Waiting to be shipped",
      "orderSent": "Shipped",
      "leadTimeNotSet": "Delivery lead time is not defined. Set it.",
      "sendBefore": "Ship before",
      "badLotSize": "Lot size not respected",
      "validateBtn": "Accept"
    }
  },
  "fr": {
    "orderModalTitle": "Commande pour",
    "editingModalTitle": "Édition de la commande pour",
    "constraintsNotMetMessage": "Certains produits ne respectent pas leurs contraintes",
    "moq": "Minimum de commande",
    "constraint": "Contrainte | Contrainte | Contraintes",
    "orderConstraints": "Contraintes de commande",
    "orderModalAddProductsButton": "Ajouter des produits",
    "orderTitle": "Commandes",
    "planned": "Prévue | Prévue | Prévues",
    "validated": "Prévalidée | Prévalidée | Prévalidées",
    "purchaseOrders": "Commande d'achat | Commande d'achat | Commandes d'achat",
    "salesOrders": "Commande de vente | Commande de vente | Commandes de vente",
    "delivered": "Livrée | Livrée | Livrées",
    "search": "Chercher par ID, fournisseur ou produit",
    "ship_before": "Expédier le",
    "send_before": "Passer commande avant le",
    "date": "Date de livraison prévue",
    "quantity": "Quantité",
    "totalQuantity": "Quantité totale",
    "addedItems": "Produits ajoutés",
    "groupedBy": "Groupé par",
    "inTotal": "Total",
    "download_selected": "Télécharger",
    "upload_selected": "Envoyer à l'API",
    "sftp_selected": "Exporter VIA SFTP",
    "sftp_exported": "Exportée via SFTP",
    "temporary_id": "ID provisoire",
    "order_id": "ID de la commande",
    "select_all": "Tout sélectionner",
    "deselect_all": "Tout désélectionner",
    "deleteSelected": "Supprimer la sélection",
    "noOrders": "Pas de commande trouvée",
    "delivery": "Livraison le",
    "constraintFrequency": "Ne respecte pas la contrainte de fréquence de commande",
    "constraintMOQ": "Ne respecte pas la contrainte de MOQ",
    "constraintFullTruck": "Ne respecte pas la contrainte de camion plein",
    "issueMOQ": " produit a un problème de MOQ ou de taille de lot | produits ont un problème de MOQ ou de taille de lot",
    "orderIssues": "Lié à la commande",
    "productIssues": "Lié aux produits",
    "days": "jours",
    "latestOrderDate": "Date de commande",
    "deliveryDate": "Date de livraison",
    "deselectAll": "Tout désélectionner",
    "suppliers": "Fournisseurs",
    "reset": "Réinitialiser",
    "dateRange": "Période",
    "thisWeek": "Cette semaine",
    "in2Weeks": "Dans 2 semaines",
    "in1month": "Dans 1 mois",
    "in2months": "Dans 2 mois",
    "in3months": "Dans 3 mois",
    "noTimeRestriction": "Pas de limitation",
    "lastOrderDate": "Date de la commande",
    "tempId": "ID provisoire",
    "orderId": "ID commande",
    "orderTitleDetail": "Détails",
    "orderTitleStatus1": "Commande prévue",
    "orderTitleStatus2": "Commande prévalidée",
    "orderTitleStatus3": "Commande d'achat",
    "orderTitleStatus4": "Commande livrée",
    "orderDetailMessageTitle": "Messages",
    "day": "jour | jour | jours",
    "value": "Valeur",
    "product": "Produit",
    "reference": "Référence",
    "lotSize": "Taille de lot",
    "leadTime": "Délai de livraion",
    "daysOfCoverage": "Couverture de stock à réception",
    "goodMoq": "Minimum de commande",
    "orderFrequency": "Fréquence de commande",
    "badFullTruckQuantity": "La contrainte de camion plein n'est pas respectée",
    "badOrderFrequency": "La contrainte de fréquence de commande n'est pas respectée",
    "fullTruckQty": "Camion plein",
    "badMOQSupply": "La quantité minimum n'est pas respectée",
    "badMOQOrder": "Le minimum de commande n'est pas respecté",
    "expectedDelivery": "Livraison attendue",
    "delete": "Supprimer",
    "cancel": "Annuler",
    "edit": "éditer",
    "cancelEdit": "annuler l'édition",
    "discardChanges": "Annuler les modifications",
    "save": "Enregistrer",
    "orderIsUpdating": "La commande est mise à jour...",
    "deleteSuccess": "Commande supprimée",
    "deleteOrderError": "Suppression impossible",
    "deleteSupplyError": "Une erreur est survenue lors de la suppression du produit",
    "deleteWarningTitle": "Confirmation de suppression",
    "deleteCurrentOrderWarningText": "Voulez-vous supprimer cette commande ?",
    "deleteCurrentOrderSupplyWarningText": "Cela va supprimer tous les approvisionnements liés à cette commande",
    "deleteSupplyWarningText": "Etes vous sûr de vouloir supprimer ce produit ?",
    "deleteWarningConfirmButton": "JE CONFIRME",
    "deleteWarningCancelButton": "ANNULER",
    "mergeWarningTitle": "Confirmation de combinaison",
    "mergeWarningText": "Une commande existe déjà pour cette date et fournisseur. Voulez-vous combiner les deux commandes?\n Ceci aura pour effet d'ajouter les produits et quantités à la commande trouvée à cette nouvelle date.",
    "mergeWarningConfirmButton": "JE CONFIRME",
    "mergeWarningCancelButton": "ANNULER",
    "secondaryUnit": "Unité secondaire",
    "orderAndSupplyConstraintNotMet": "Certaines contraintes au niveau commande et au niveau produit ne sont pas respectées",
    "orderConstraintNotMet": "Certaines contraintes au niveau commande ne sont pas respectées",
    "supplyConstraintNotMet": "Certains produits ne respectent pas leurs contraintes",
    "customer": {
      "display": "Client",
      "partner": "Fournisseur",
      "waitValidateOrder": "En attente de validation",
      "orderValidated": "Validé",
      "waitSendOrder": "En attente d'expédition",
      "sendBefore": "Expédier avant",
      "orderSent": "Expédié",
      "badLotSize": "La taille du lot n'est pas respectée",
      "validateBtn": "Valider"
    },
    "supplier": {
      "display": "Fournisseur",
      "partner": "Client",
      "waitValidateOrder": "En attente de validation",
      "orderValidated": "Accepté",
      "waitSendOrder": "En attente d'expédition",
      "leadTimeNotSet": "Le délai de livraison n'est pas défini. Configurer.",
      "sendBefore": "Expédier avant",
      "orderSent": "Expédié",
      "badLotSize": "La taille du lot n'est pas respectée",
      "validateBtn": "Accepter"
    }
  }
}
