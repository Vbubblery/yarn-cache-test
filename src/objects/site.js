import { validate } from "jsonschema";
import SiteSchema from "@/schemas/site";

const DEFAULT_HORIZON_WEEKS = 52;
export class Site {
  constructor(model) {
    this._id = model.id;
    this._name = model.name;
    this._address = model.address;
    this._company = model.company;
    this._users = model.users ? model.users.map(user => user.id) : [];
    this._storages = model.storage_sites;
    this._isSupplierView = model.is_supplier_view ?? false;
    this._demandDistributionWeek = model.demand_distribution_week;
    this._currentYearDemand = model.current_year_demand;
    this._currentYearMonthlyStock = model.current_year_monthly_stock;
    this._currentYearTotalValues = model.current_year_total_values;
    this._avgSiteStockouts = model.avg_site_stockout;
    this._hasSameDefaultUnit = model.has_same_default_unit;
    this._showFutureDashboard = model.show_future_dashboard;
    this._currency = model.currency ?? "eur";
    this._aggSiteAccuracy = model.agg_site_accuracy;
    this._wAvgSiteAccuracy = model.w_avg_site_accuracy;
    this._avgSiteAccuracy = model.avg_site_accuracy;
    this._hasShelfLife = model.has_shelf_life;
    this._displayDemandForecastExternal =
      model.display_demand_forecast_external;
    this._displayFirmOrdersInDemand = model.display_firm_orders_in_demand;
    this._hasStockWithBacklog = model.has_stock_with_backlog;
    this._defaultDemandTimeBucket = model.default_demand_time_bucket;
    this._defaultPlanningTimeBucket = model.default_planning_time_bucket;
    this._currentYearMonthlyStockCoverage =
      model.current_year_monthly_stock_coverage;
    this._planningHorizonWeeks =
      model.global_config?.[0]?.planning_horizon_week ?? DEFAULT_HORIZON_WEEKS;
    this._demandHorizonWeeks =
      model.global_config?.[0]?.demand_horizon_week ?? DEFAULT_HORIZON_WEEKS;
  }
  static create(model) {
    return validate(model, Site.schema()) ? new Site(model) : null;
  }
  static schema() {
    return SiteSchema;
  }

  // Getters
  get id() {
    return this._id;
  }
  get name() {
    return this._name;
  }
  get address() {
    return this._address;
  }
  get users() {
    return this._users;
  }
  get storages() {
    return this._storages;
  }
  get company() {
    return this._company;
  }
  get isSupplierView() {
    return this._isSupplierView;
  }
  get demandDistributionWeek() {
    return this._demandDistributionWeek;
  }
  get currentYearDemand() {
    return this._currentYearDemand;
  }
  get currentYearMonthlyStock() {
    return this._currentYearMonthlyStock;
  }
  get currentYearMonthlyStockCoverage() {
    return this._currentYearMonthlyStockCoverage;
  }
  get currentYearTotalValues() {
    return this._currentYearTotalValues;
  }
  get avgSiteStockouts() {
    return this._avgSiteStockouts;
  }
  get showFutureDashboard() {
    return this._showFutureDashboard;
  }
  get currency() {
    return this._currency;
  }
  get hasSameDefaultUnit() {
    return this._hasSameDefaultUnit;
  }
  get aggSiteAccuracy() {
    return this._aggSiteAccuracy;
  }
  get wAvgSiteAccuracy() {
    return this._wAvgSiteAccuracy;
  }
  get avgSiteAccuracy() {
    return this._avgSiteAccuracy;
  }
  get hasShelfLife() {
    return this._hasShelfLife;
  }
  get hasStockWithBacklog() {
    return this._hasStockWithBacklog;
  }
  get displayDemandForecastExternal() {
    return this._displayDemandForecastExternal;
  }
  get displayFirmOrdersInDemand() {
    return this._displayFirmOrdersInDemand;
  }

  get defaultDemandTimeBucket() {
    return this._defaultDemandTimeBucket;
  }

  get defaultPlanningTimeBucket() {
    return this._defaultPlanningTimeBucket;
  }

  get planningHorizonWeeks() {
    return this._planningHorizonWeeks;
  }
  get demandHorizonWeeks() {
    return this._demandHorizonWeeks;
  }
  // Setters
  set id(id) {
    this._id = id;
  }
  set name(name) {
    this._name = name;
  }
  set address(address) {
    this._address = address;
  }
  set users(users) {
    this._users = users;
  }
  set storages(storages) {
    this._storages = storages;
  }
  set company(company) {
    this._company = company;
  }
  set isSupplierView(isSupplierView) {
    this._isSupplierView = isSupplierView;
  }
  set demandDistributionWeek(demandDistributionWeek) {
    this._demandDistributionWeek = demandDistributionWeek;
  }
  set currentYearDemand(currentYearDemand) {
    this._currentYearDemand = currentYearDemand;
  }
  set currentYearMonthlyStock(currentYearMonthlyStock) {
    this._currentYearMonthlyStock = currentYearMonthlyStock;
  }
  set currentYearTotalValues(currentYearTotalValues) {
    this._currentYearTotalValues = currentYearTotalValues;
  }
  set avgSiteStockouts(avgSiteStockouts) {
    this._avgSiteStockouts = avgSiteStockouts;
  }
  set showFutureDashboard(showFutureDashboard) {
    this._showFutureDashboard = showFutureDashboard;
  }
  set currency(currency) {
    this._currency = currency;
  }
  set hasSameDefaultUnit(hasSameDefaultUnit) {
    this._hasSameDefaultUnit = hasSameDefaultUnit;
  }
  set aggSiteAccuracy(aggSiteAccuracy) {
    this._aggSiteAccuracy = aggSiteAccuracy;
  }
  set wAvgSiteAccuracy(wAvgSiteAccuracy) {
    this._wAvgSiteAccuracy = wAvgSiteAccuracy;
  }
  set avgSiteAccuracy(avgSiteAccuracy) {
    this._avgSiteAccuracy = avgSiteAccuracy;
  }
  set hasShelfLife(hasShelfLife) {
    this._hasShelfLife = hasShelfLife;
  }
  set hasStockWithBacklog(hasStockWithBacklog) {
    this._hasStockWithBacklog = hasStockWithBacklog;
  }
  set displayDemandForecastExternal(displayDemandForecastExternal) {
    this._displayDemandForecastExternal = displayDemandForecastExternal;
  }
  set displayFirmOrdersInDemand(displayFirmOrdersInDemand) {
    this._displayFirmOrdersInDemand = displayFirmOrdersInDemand;
  }

  set defaultDemandTimeBucket(timeBucket) {
    this._defaultDemandTimeBucket = timeBucket;
  }
  set defaultPlanningTimeBucket(timeBucket) {
    this._defaultPlanningTimeBucket = timeBucket;
  }
  set planningHorizonWeeks(months) {
    this._planningHorizonWeeks = months;
  }
  set demandHorizonWeeks(months) {
    this._demandHorizonWeeks = months;
  }
}
