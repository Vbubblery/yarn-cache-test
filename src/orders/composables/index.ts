export { useOrdersSearch } from "./ordersSearch";
export { useOrderDetails } from "./orderDetails";
export { useOrderActions } from "./orderActions";
