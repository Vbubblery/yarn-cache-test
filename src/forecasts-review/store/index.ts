import { computed, SetupContext, ComputedRef } from "@vue/composition-api";
import { PLANNING_DEMAND_VIEW } from "@/forecasts-review/typings";
import { Site } from "@/objects/site";

export const useBaseResources = (context: SetupContext) => {
  const store = context.parent?.$store || null;

  const getMe = computed<{ locale: string }>(
    () => store?.getters["common/getMe"]
  );
  const planningDemandView: ComputedRef<PLANNING_DEMAND_VIEW> = computed(
    () => store?.getters["common/getDefaultView"]
  );
  const currentSite = computed<Site>(
    () => store?.getters["common/getCurrentSite"]
  );
  const tablesData = computed(() => store?.getters["demand/getTablesData"]);
  const demandGetChartData = computed(
    () => store?.getters["demand/getChartData"]
  );
  const demandGetIsLoadingChart = computed(
    () => store?.getters["demand/getIsLoadingChart"]
  );
  const demandGetIsLoadingTables = computed(
    () => store?.getters["demand/getIsLoadingTables"]
  );
  const planningGetChartData = computed(
    () => store?.getters["planning/getChartData"]
  );
  const planningGetIsLoadingChart = computed(
    () => store?.getters["planning/getIsLoadingChart"]
  );
  const planningGetIsLoadingTables = computed(
    () => store?.getters["planning/getIsLoadingTables"]
  );
  const selectedTagIds = computed(
    () => store?.getters["common/getSelectedTags"]
  );
  const selectedCustomerId = computed(() =>
    currentSite.value.isSupplierView
      ? store?.getters["demand/getSelectedCustomerSiteId"]
      : null
  );
  const updateDefaultView = (view: PLANNING_DEMAND_VIEW) => {
    store?.commit("common/updateDefaultView", view);
  };
  const updateSelectedProducts = (items: object[]) => {
    store?.dispatch("common/updateSelectedProducts", items);
  };
  const updateSelectedTags = (items: object[]) => {
    store?.dispatch("common/updateSelectedTags", items);
  };
  const updateDataType = (itemType: string) => {
    store?.commit("common/updateDataType", itemType);
  };
  const resetDemandState = () => {
    store?.commit("demand/resetState");
  };
  const updateDemandPage = (pageNum: number) => {
    store?.commit("demand/updatePage", pageNum);
  };
  const updateDemandTableOnly = (onlyTable: boolean) => {
    store?.commit("demand/updateOnlyTable", onlyTable);
  };
  const fetchData = (
    selectedView: undefined | PLANNING_DEMAND_VIEW = undefined
  ) => {
    const currentView = selectedView ?? planningDemandView.value;
    store?.dispatch(
      currentView === PLANNING_DEMAND_VIEW.DEMAND
        ? "demand/fetchDemandData"
        : "planning/updatePlanningData"
    );
  };
  const updateSelectedCustomerSiteId = (customerSiteId: number | null) => {
    store?.commit("demand/updateSelectedCustomerSiteId", customerSiteId);
  };
  return {
    getMe,
    planningDemandView,
    updateDefaultView,
    currentSite,
    tablesData,
    demandGetChartData,
    demandGetIsLoadingChart,
    demandGetIsLoadingTables,
    planningGetChartData,
    planningGetIsLoadingChart,
    planningGetIsLoadingTables,
    selectedCustomerId,
    selectedTagIds,
    updateSelectedProducts,
    updateSelectedTags,
    updateDataType,
    resetDemandState,
    updateDemandPage,
    updateDemandTableOnly,
    fetchData,
    updateSelectedCustomerSiteId
  };
};
