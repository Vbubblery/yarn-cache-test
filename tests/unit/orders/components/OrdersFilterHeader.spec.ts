import { renderComponent } from "tests/test-utils";
import OrdersFilterHeader from "@/orders/components/OrdersFilterHeader.vue";
// eslint-disable-next-line jest/no-mocks-import
import { store } from "@/store/__mocks__/store";
import { fireEvent, waitFor } from "@testing-library/vue";
import { useOrdersSearch } from "@/orders/composables";
import { ORDER_TYPE_VALUE } from "@/orders/constants";
import { Site } from "@/objects/site";
import * as OrdersBackend from "@/orders/backend";
import SupervisorService from "@/services/supervisor.service";
jest.mock("@/helper/validation");

const orderStatus: { [key: string]: string } = {
  [ORDER_TYPE_VALUE.PLANNED]: "PLANNED",
  [ORDER_TYPE_VALUE.VALIDATED]: "VALIDATED",
  [ORDER_TYPE_VALUE.PURCHASE_ORDER]: "PURCHASE_ORDER",
  [ORDER_TYPE_VALUE.DELIVERED]: "DELIVERED"
};

const fetchOrders = jest
  .spyOn(SupervisorService, "get")
  .mockResolvedValue(jest.fn());
const fetchOrdersCountByStatus = jest
  .spyOn(OrdersBackend, "fetchOrdersCountByStatus")
  .mockResolvedValue({
    all: 1,
    1: 1,
    2: 0,
    3: 0,
    4: 0
  });

describe("Orders Filter Header", () => {
  beforeEach(() => {
    fetchOrders.mockClear();
    fetchOrdersCountByStatus.mockClear();
  });
  it("renders the orders' filter header", () => {
    const { html } = renderComponent(OrdersFilterHeader, {}, () => ({
      store
    }));
    expect(html()).toMatchSnapshot();
  });
  it("updates search keyword on input", async () => {
    const newSearch = "newSearch";
    const { searchKeyword } = useOrdersSearch();
    const { findByTestId } = renderComponent(OrdersFilterHeader, {}, () => ({
      store
    }));
    const searchInput = await findByTestId(/search_orders_page/i);
    await fireEvent.update(searchInput, newSearch);

    //wait for the debounce to be effective
    setTimeout(() => expect(searchKeyword.value).toEqual(newSearch), 500);
  });

  it.each(Object.entries(orderStatus))(
    "updates selected status %s named %s on filter click",
    async key => {
      const { selectedOrderType } = useOrdersSearch();
      const { getByTestId } = renderComponent(OrdersFilterHeader, {}, () => ({
        store
      }));
      const filterButton = getByTestId(new RegExp(`order_status_${key}_tab`));
      fireEvent
        .click(filterButton)
        .then(() => expect(selectedOrderType.value).toEqual(+key));
    }
  );

  it("fetches new count of orders by status on site change", async () => {
    renderComponent(OrdersFilterHeader, {}, () => ({
      store
    }));
    store.commit(
      "common/initCurrentSite",
      Site.create({
        id: 2,
        name: "TestSite",
        address: "42, rue du FooBar",
        currency: "eur",
        company: 2,
        users: [{ id: "2" }],
        storage_sites: [{ id: "1", name: "storageTest" }],
        demand_distribution_week: [1, 0, 0, 0, 0, 0, 0],
        customer_sites: [],
        planning_horizon_months: 12,
        demand_horizon_months: 12
      })
    );
    await waitFor(() =>
      // The fetch is called on component creation and by watcher trigger
      expect(fetchOrdersCountByStatus).toHaveBeenCalledTimes(2)
    );
  });
});
