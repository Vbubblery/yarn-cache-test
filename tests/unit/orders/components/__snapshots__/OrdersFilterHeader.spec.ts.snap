// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Filter Header renders the orders' filter header 1`] = `
<div>
  <div class="flex px-6 py-4 h-full mx-2 my-5">
    <div class="font-montserrat leading-6 text-2xl font-normal mx-4 my-1 mr-12">
      orderTitle
    </div>
    <div class="flex"><button class="mx-2 
    rounded-[30px]
    font-montserrat
    font-medium
    normal
    h-full
    py-1
    justify-between
    
   
    px-6 inline-block align-middle
    
    text-gray-600
    group
    hover:bg-blue-100  hover:text-blue-700 focus:bg-blue-100 focus:text-blue-700 active:bg-blue-200 active:text-blue-700
    space-x-2
    items-center
    " data-testid="order_status_1_tab">
        <div class="py-1  text-lg inline-block align-middle">
          planned
        </div>
        <!---->
      </button><button class="mx-2 
    rounded-[30px]
    font-montserrat
    font-medium
    normal
    h-full
    py-1
    justify-between
    
   
    px-6 inline-block align-middle
    
    text-gray-600
    group
    hover:bg-blue-100  hover:text-blue-700 focus:bg-blue-100 focus:text-blue-700 active:bg-blue-200 active:text-blue-700
    space-x-2
    items-center
    " data-testid="order_status_2_tab">
        <div class="py-1  text-lg inline-block align-middle">
          validated
        </div>
        <!---->
      </button><button class="mx-2 
    rounded-[30px]
    font-montserrat
    font-medium
    normal
    h-full
    py-1
    justify-between
    
   
    px-6 inline-block align-middle
    
    text-gray-600
    group
    hover:bg-blue-100  hover:text-blue-700 focus:bg-blue-100 focus:text-blue-700 active:bg-blue-200 active:text-blue-700
    space-x-2
    items-center
    " data-testid="order_status_3_tab">
        <div class="py-1  text-lg inline-block align-middle">
          purchaseOrders
        </div>
        <!---->
      </button><button class="mx-2 
    rounded-[30px]
    font-montserrat
    font-medium
    normal
    h-full
    py-1
    justify-between
    
   
    px-6 inline-block align-middle
    
    text-gray-600
    group
    hover:bg-blue-100  hover:text-blue-700 focus:bg-blue-100 focus:text-blue-700 active:bg-blue-200 active:text-blue-700
    space-x-2
    items-center
    " data-testid="order_status_4_tab">
        <div class="py-1  text-lg inline-block align-middle">
          delivered
        </div>
        <!---->
      </button></div>
  </div>
  <div class="h-12 border-b grid grid-cols-12 items-center shadow-md">
    <div class="flex items-center h-full col-span-4">
      <div class="h-full flex items-center justify-center" style="width: 80px;"><svg viewBox="0 0 24 24" width="24" height="24" class="icon">
          <use xlink:href="/assets/svg/sprite.svg#svg-search-line" href="/assets/svg/sprite.svg#svg-search-line"></use>
        </svg></div>
      <div class="flex justify-start w-full p-1 rounded border-r" style="height: 90%;">
        <div class="flex items-center w-full h-full bg-white border focus-within:border-blue-400 rounded"><input placeholder="search..." data-testid="search_orders_page" class="leading-8 rounded-lg px-2 w-full"></div>
      </div>
    </div>
    <div class="flex items-center h-full w-full col-span-8">
      <div class="flex items-center h-full col-span-2">
        <section class="dropDownMenuWrapper h-full  dropdown cursor-pointer">
          <div class="dropDownMenuButton px-4 ant-dropdown-trigger" data-testid="base_dropdown">
            <div class="w-[90%]">
              <p class="text-gray-500 text-xs select-none">suppliers</p>
              <p class="text-blue-600 truncate">

              </p>
            </div> <svg viewBox="0 0 24 24" width="24" height="24" class="icon">
              <use xlink:href="/assets/svg/sprite.svg#svg-arrow-down-s-line" href="/assets/svg/sprite.svg#svg-arrow-down-s-line"></use>
            </svg>
          </div>
        </section>
      </div>
      <div class="flex items-center h-full col-span-2 px-2">
        <section class="dropDownMenuWrapper h-full max-w-6xl dropdown cursor-pointer border-l">
          <div class="dropDownMenuButton px-4 ant-dropdown-trigger" data-testid="base_dropdown">
            <div class="w-[90%]">
              <p class="text-gray-500 text-xs select-none">dateRange</p>
              <p class="text-blue-600 truncate">

              </p>
            </div> <svg viewBox="0 0 24 24" width="24" height="24" class="icon">
              <use xlink:href="/assets/svg/sprite.svg#svg-arrow-down-s-line" href="/assets/svg/sprite.svg#svg-arrow-down-s-line"></use>
            </svg>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
